### PR TITLE
Add more logging to help users understand why libre 2 does not communicate with bluetooth.

### PR DESCRIPTION
**Pull request by tzachi-dar**
I have received a few complaints that xDrip does not work with libre2.
Usually this things are solved when the user retries the connection a few times.
This change high level logging weather scanning the sensor and enabling BT has worked correctly, in order to make users know whether they enabled the sensor successfully or not.

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java [hunk 1]
```diff
@@ -474,13 +480,27 @@ protected void onPostExecute(Tag tag) {
             }
         }
 
+        void LogLibre2StartStreeming(ENABLE_STREAMING success, String extraData) {
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java:483]: Spelling typo, I assume this should be streaming not streeming

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java [hunk 2]
```diff
@@ -142,13 +142,6 @@ public synchronized static void cleanupRaw() {
     }
 
 
-    public static List<UserError> all() {
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java:145]: any particular reason to remove this? proguard will strip unused methods from the apk automatically.

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java (hunk 1 above)
Comment (tzachi-dar) [reply app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java]: Fixed.

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java (hunk 2 above)
Comment (tzachi-dar) [reply app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java:145]: Well, I thought that code is easier to read if there are no unused functions.
I Can return this if you think it has value.

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java [hunk 3]
```diff
@@ -492,6 +512,7 @@ void startLibre2Streaming(NfcV nfcvTag, byte[] patchUid, byte[] patchInfo) throw
             Pair<byte[], String> unlockData = LibreOOPAlgorithm.nfcSendgetBluetoothEnablePayload();
             if (unlockData == null) {
                 Log.e(TAG, "unlockData is null, not enabeling streaming");
+                LogLibre2StartStreaming(ENABLE_STREAMING.FAILED, "Failure in communicating with OOP2. Is OOP2 istalled");
```
Comment (anbraten) [inline app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java:515]:
```suggestion
                LogLibre2StartStreaming(ENABLE_STREAMING.FAILED, "Failure in communicating with OOP2. Is OOP2 installed?");
```

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java [hunk 4]
```diff
@@ -474,13 +480,27 @@ protected void onPostExecute(Tag tag) {
...
+        }
+
         void startLibre2Streaming(NfcV nfcvTag, byte[] patchUid, byte[] patchInfo) throws InterruptedException {
             // Since this is libre2 we can remove the physical devices battery.
             Pref.setInt("bridge_battery", 0);
             PersistentStore.setString("Tomatobattery", "0");
             PersistentStore.setString("Bubblebattery", "0");
             if (!enableBluetoothAllowed(context)) {
                 Log.e(TAG, "Sensor is libre 2, enabeling BT not allowed");
+                LogLibre2StartStreaming(ENABLE_STREAMING.FAILED, "Sensor is Libre 2 but enabling BT not allowed in settings");
```
Comment (anbraten) [inline app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java:503]:
```suggestion
                LogLibre2StartStreaming(ENABLE_STREAMING.FAILED, "Sensor is Libre 2 but enabling BT is not allowed by the settings");
```

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java [hunk 5]
```diff
@@ -474,13 +480,27 @@ protected void onPostExecute(Tag tag) {
...
         }
 
+        void LogLibre2StartStreaming(ENABLE_STREAMING success, String extraData) {
+            switch(success) {
+                case SUCCESS:
+                    UserError.Log.ueh("Libre 2", "bluetooth connection with sensor enabled, you should be able to receive BG data using bluetooth");
+                    break;
+                case FAILED:
+                    UserError.Log.ueh("Libre 2", "bluetooth connection with sensor failed." +
+                            " You will not have readings. Try scanning again to fix this. (Extra data: " +
```
Comment (anbraten) [inline app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java:490]:
```suggestion
                            " You wont get any readings. Try scanning again to fix this. (Extra data: " +
```

File: app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java [hunk 6]
```diff
@@ -474,13 +480,27 @@ protected void onPostExecute(Tag tag) {
             }
         }
 
+        void LogLibre2StartStreaming(ENABLE_STREAMING success, String extraData) {
```
Comment (tolot27) [inline app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java:483]: Wouldn't a simple `boolean` value be sufficient here, would it? Or do you plan to extend the enum in the future to reflect fine granular streaming states?

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java (hunk 2 above)
Comment (tolot27) [reply app/src/main/java/com/eveningoutpost/dexdrip/Models/UserError.java:145]: Please keep it for now.

**Review by tolot27 — CHANGES_REQUESTED** [review]
This PR is now cluttered with many "fix" commits and I am one of those who make it worse. Please reset the branch and merge them into one. Or simply don't pull this branch but fix the changes in your latest commit 25b8a0e and do a "git commit --amend" and "git push --force-with-lease" afterwards. That should resolve the clutter.

**Comment by tzachi-dar** [comment]
I have returned the missing functions, ammended the typos.
Using an enum and not a bool as it makes the code more readable.

Please reivew again.

File: app/src/main/java/com/eveningoutpost/dexdrip/ErrorsActivity.java [hunk 7]
```diff
@@ -27,6 +27,7 @@
 
 /**
  * Created by Emma Black on 8/3/15.
+ * This is the old logs activity.
```
Comment (tolot27) [inline app/src/main/java/com/eveningoutpost/dexdrip/ErrorsActivity.java:30]: What does "old logs" mean here? Is the class `ErrorsActivity` obsolete? If yes, we should mark it as deprecated and remove it in future versions.

**Review by tolot27 — APPROVED** [review]
LGTM

File: app/src/main/java/com/eveningoutpost/dexdrip/ErrorsActivity.java (hunk 7 above)
Comment (tzachi-dar) [reply app/src/main/java/com/eveningoutpost/dexdrip/ErrorsActivity.java:30]: If you look at the ui of the error logs, and press the menu on the right there is an option there: "view old error log".
It is a similar screen, but looks (and implemented) differently. I'm not sure how many people are still using this.
Maybe it is time to remove it.

**Review by jamorham — APPROVED** [review]

**Pull request merged**